caper/cromwell_metadata.py

```diff
@@ -70,6 +70,30 @@ def workflow_id(self):
     def workflow_status(self):
         return self._metadata.get('status')
 
+    @property
+    def workflow_root(self):
+        if 'workflowRoot' in self._metadata:
+            return self._metadata['workflowRoot']
+        else:
+            call_roots = []
+
+            def get_call_root(call_name, call, parent_call_names):
+                nonlocal call_roots
+                call_root = call.get('callRoot')
+                if call_root:
+                    call_roots.append(call_root)
+
+            self.recurse_calls(get_call_root)
```

Comment:
Rather than creating a callback that requires a `nonlocal`, it might make sense to have a function that returns the list of already recursed calls (could make it an`@property` on the instance). Then this function would be much simpler.